expertise/service/utils.py

```diff
@@ -1219,71 +1214,49 @@ def _get_scores_and_metadata_streaming(all_blobs, job_id, group_scoring=False, p
             metadata = json.loads(metadata_files[0].download_as_string())
             yield {'metadata': metadata, 'results': []}
 
-            # Then stream the scores
-            if not skip_sparse:
-                sparse_score_files = [
-                    blob for blob in score_files if 'sparse' in blob.name
-                ]
-                if len(sparse_score_files) != 1:
-                    raise openreview.OpenReviewException(f"Internal Error: incorrect sparse score files found expected [1] found {len(sparse_score_files)}")
-                
-                # Stream the sparse scores in chunks
-                downloader = sparse_score_files[0].open('r')
+            # Helper function to stream scores from a blob file
+            def stream_score_file(blob):
+                downloader = blob.open('r')
                 chunk = downloader.readline()
                 results_chunk = []
                 chunk_size = 1000  # Adjust this number based on your data size
-                
+
                 while chunk:
                     if isinstance(chunk, bytes):
                         chunk = chunk.decode('utf-8')
                     if chunk.strip():
                         results_chunk.append(json.loads(chunk.strip()))
-                    
+
                     # Yield chunks of results
                     if len(results_chunk) >= chunk_size:
                         yield {'results': results_chunk, 'metadata': None}
                         results_chunk = []
-                    
+
                     chunk = downloader.readline()
-                
+
                 # Yield any remaining results
                 if results_chunk:
                     yield {'results': results_chunk, 'metadata': None}
-                    
+
                 downloader.close()
+
+            # Determine whether to use sparse or non-sparse score files
+            sparse_score_files = [
```

Comment:
Why do we have an array of files? is there a chance we need to stream more than one file?

Reply:
I don't think that's ever supposed to happen. I think @haroldrubio has it as some form of validation. If there's more than one file with the `sparse` substring, it fails. I am not sure why this would happen though. @haroldrubio, is my interpretation correct?

Reply:
I don't think it's supposed to happen, we can filter by "_sparse" or maybe build the filename directly? I think I implemented it this way because I wasn't sure what the name of the file was going to be but I think it should be easier now